app/deployments/routes.py

```diff
@@ -1420,17 +1500,141 @@ def createdep():
     else:
         remove_sla_from_template(template)
 
+    create_dep_method(
+        source_template,
+        selected_template,
+        additionaldescription,
+        inputs,
+        form_data,
+        template,
+        template_text,
+    )
+
+    return redirect(url_for(SHOW_DEPLOYMENTS_ROUTE))
+
+
+@deployments_bp.route("/<depid>/retry")
+@auth.authorized_with_valid_token
+def retrydep(depid=None):
+    """
+    A function to retry a failed deployment.
+    Parameters:
+    - depid: str, the ID of the deployment
+    """
+    tosca_info, _, _ = tosca.get()
+    access_token = iam.token["access_token"]
+
+    # retrieve deployment from DB
+    dep = dbhelpers.get_deployment(depid)
+
+    if dep is None or dep.selected_template == "":
+        flash(
+            "The selected deployment is invalid. Try creating it from scratch.",
+            "danger",
+        )
+        return redirect(url_for(SHOW_DEPLOYMENTS_ROUTE))
+
+    inputs = process_deployment_data(dep)[0]
+
+    # Get the max num retry for the new name
+    max_num_retry = 0
+    str_retry = " retry_"
+    tmp_name = dep.description.split(str_retry)[0]
+
+    group = None
+    if "active_usergroup" in session and session["active_usergroup"] is not None:
+        group = session["active_usergroup"]
+
+    deployments = []
+    try:
+        deployments = app.orchestrator.get_deployments(
+            access_token, created_by="me", user_group=group
+        )
+    except Exception as e:
+        flash("Error retrieving deployment list: \n" + str(e), "warning")
+
+    if deployments:
```

Comment:
Quite bizarre to write only one line inside the `try except` block. Maybe but everything that needs the `deployments` variable inside the `try/except`, or consider to move it inside the `app.orchestrator.get_deployment` method